libraries/AP_HAL/Semaphores.h

```diff
@@ -2,7 +2,7 @@
 
 #include "AP_HAL_Namespace.h"
 
-#define HAL_SEMAPHORE_BLOCK_FOREVER ((uint32_t) 0xFFFFFFFF)
+#define HAL_SEMAPHORE_BLOCK_FOREVER 0
```

Comment:
Great. And after this big refactor let's rename Semaphore to Mutex since this is what it actually is? :)